src/main/java/io/debezium/connector/yugabytedb/YugabyteDBSnapshotChangeEventSource.java

```diff
@@ -98,6 +103,7 @@ public YugabyteDBSnapshotChangeEventSource(YugabyteDBConnectorConfig connectorCo
         this.tabletSafeTime = new HashMap<>();
         this.tableIdToTable = new HashMap<>();
         this.shouldWaitForCallback = new HashMap<>();
+        this.tabletsWaitingForCallback = new HashSet<>();
```

Comment:
Where do we add/remove from this set?

Reply:
Add here once we detect that this tablet hasn't caught up on the explicit checkpoint: https://github.com/yugabyte/debezium-connector-yugabytedb/blob/main/src/main/java/io/debezium/connector/yugabytedb/YugabyteDBSnapshotChangeEventSource.java#L559

Remove once we receive the callback and know that the tablet has completed the snapshot: https://github.com/yugabyte/debezium-connector-yugabytedb/blob/main/src/main/java/io/debezium/connector/yugabytedb/YugabyteDBSnapshotChangeEventSource.java#L646